ci.yml

```diff
@@ -67,23 +67,33 @@ jobs:
                 params:
                     submodules: [build/terraform]
               - get: ci
-            # TODO(@ndmckinley): This task is too monolithic and should be split up.
-          - task: generate
-            file: ci/magic-modules/generate.yml
-            params:
-                GH_USERNAME: ((username))
+            # consumes: ci, magic-modules (detached HEAD)
+            # produces: magic-modules-branched (new branch, with submodule)
+          - task: branch-magic-modules
+            file: ci/magic-modules/branch.yml
+            # consumes: ci, magic-modules-branched
+            # produces: terraform-generated
+          - task: generate-terraform
+            file: ci/magic-modules/generate-terraform.yml
+            # Puts 'terraform-generated' into the robot's fork.
           - put: terraform-intermediate
             params:
-                repository: mm-output/build/terraform
-                branch_file: mm-output/branchname
+                repository: terraform-generated
+                branch_file: magic-modules-branched/branchname
                 only_if_diff: true
-
-          # This needs to go below all the submodules because it has a commit which
+            # consumes: ci, magic-modules-branched
+            # produces: magic-modules-submodules
+          - task: point-to-submodules
+            file: ci/magic-modules/point-to-submodules.yml
+            params:
+              # This needs to match the username for the 'terraform-intermediate' resource.
+              GH_USERNAME: ((username))
+          # This 'put' needs to go below all the submodules because it has a commit which
```

Comment:
To avoid relying on a comment to make sure somebody does not create a broken config, is it possible to have some "subtask" or "dependent job" that would run after all submodule tasks are complete? I mean, in a way that one could not easily break if this comment is not spotted?

Reply:
There's not really a mechanism for that - or, more accurately, the mechanism for keeping that from happening is the task ordering itself.  It would be an obvious error to write that hypothetical broken config, because you'd be writing a config that made some changes but never pushed them.  You'd see it immediately, and if you didn't, it'd be obvious when you went to go test.  I don't know of any mechanism for warning you about that kind of mistake.